command_lib/base.yml

```diff
@@ -87,3 +87,34 @@ debian:
           delimiter: 'EOF'
         src_urls: {}
 
+# openjdk -------------------------------------------------------------------------------------------------------------------------------
+openjdk:
+  latest: '8'
+  tags:
+    '8':
+        shell: '/bin/bash'
+        # either enter the package names as a list or a snippet to invoke either within or outside the container
+        names:
+          invoke:
+            1:
+              # Currently two types of environments are supported - host or container
+              container:
+                - "dpkg --get-selections | cut -f1 -d':' | awk '{print $1}'"
+          delimiter: "\n"
+        # the length of this list should be the same as the names or else there will be a mismatch
+        versions:
+          invoke:
+            1:
+              container:
+                - "pkgs=`dpkg --get-selections | cut -f1 -d':' | awk '{print $1}'`"
+                - "for p in $pkgs; do dpkg -l $p | awk 'NR>5 {print $3}'; done"
+          delimiter: "\n"
+        licenses:
+          invoke:
+            1:
+              container:
+                - "pkgs=`dpkg --get-selections | cut -f1 -d':' | awk '{print $1}'`"
+                - "for p in $pkgs; do cat /usr/share/doc/$p/copyright; echo EOF; done"
```

Comment:
Welcome back! The snippet for obtaining the licenses doesn't produce the same number of results as the actual packages that are installed:
python verify_invoke.py --container testcontainer --keys base openjdk tags 8 names
['adduser', 'adwaita-icon-theme', 'apt', 'at-spi2-core', 'base-files', ...
...
331 <-- total number of packages installed in the container
python verify_invoke.py --container testcontainer --keys base openjdk tags 8 licenses
......
390 <-- more values than number of packages

The tool will not put in any values for licenses in the report because of the mismatch. 'EOF' may not be a good delimiter in this case, so maybe try to modify the script "for p in $pkgs; do cat /usr/share/doc/$p/copyright; echo EOF; done" so you can get 331 values

Reply:
What is the command line to run verify_invoke.py so I can do the verification on my side?  The openjdk:8 is a cut and paste of debian:jessie for the base.yml.  So isn't debian:jessie going to run into the same issue with the package count?

Reply:
python verify_invoke.py --container testcontainer --keys base openjdk tags 8 licenses

the keys are the keys in the yaml file uptil the 'invoke' part. You can test this out with 'names', 'versions', 'licenses' and 'src_urls'.

The number of results for debian:jessie match. You can test it out on a container running debian:jessie and use the same command above on it.

Reply:
I get 331 results if I use the delimited 'ENDOFFILE'

licenses:
          invoke:
            1:
              container:
                - "pkgs=`dpkg --get-selections | cut -f1 -d':' | awk '{print $1}'`"
                - "for p in $pkgs; do cat /usr/share/doc/$p/copyright; echo ENDOFFILE; done"
          delimiter: 'ENDOFFILE'